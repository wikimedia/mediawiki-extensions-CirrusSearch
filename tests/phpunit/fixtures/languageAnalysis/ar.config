{
	"extended_tests":true,
	"expected.lang":"ar.core"
}
