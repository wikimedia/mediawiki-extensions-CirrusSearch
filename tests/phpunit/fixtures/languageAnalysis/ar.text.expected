{
    "analyzer": {
        "keyword": {
            "filter": [
                "truncate_keyword"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "lowercase_keyword": {
            "filter": [
                "truncate_keyword",
                "lowercase"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "near_match": {
            "char_filter": [
                "near_space_flattener"
            ],
            "filter": [
                "lowercase"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "near_match_asciifolding": {
            "char_filter": [
                "near_space_flattener"
            ],
            "filter": [
                "truncate_keyword",
                "lowercase",
                "asciifolding"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "plain": {
            "char_filter": [
                "nnbsp_norm",
                "word_break_helper"
            ],
            "filter": [
                "icutokrep_no_camel_split",
                "lowercase"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        },
        "plain_search": {
            "char_filter": [
                "nnbsp_norm",
                "word_break_helper"
            ],
            "filter": [
                "icutokrep_no_camel_split",
                "lowercase"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        },
        "prefix": {
            "char_filter": [
                "near_space_flattener"
            ],
            "filter": [
                "lowercase"
            ],
            "tokenizer": "prefix",
            "type": "custom"
        },
        "prefix_asciifolding": {
            "char_filter": [
                "near_space_flattener"
            ],
            "filter": [
                "lowercase",
                "asciifolding"
            ],
            "tokenizer": "prefix",
            "type": "custom"
        },
        "short_text": {
            "filter": [
                "lowercase",
                "aggressive_splitting",
                "asciifolding_preserve",
                "dedup_asciifolding"
            ],
            "tokenizer": "whitespace",
            "type": "custom"
        },
        "short_text_search": {
            "filter": [
                "lowercase",
                "aggressive_splitting"
            ],
            "tokenizer": "whitespace",
            "type": "custom"
        },
        "source_text_plain": {
            "char_filter": [
                "word_break_helper_source_text"
            ],
            "filter": [
                "icutokrep_no_camel_split",
                "lowercase"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        },
        "source_text_plain_search": {
            "char_filter": [
                "word_break_helper_source_text"
            ],
            "filter": [
                "icutokrep_no_camel_split",
                "lowercase"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        },
        "suggest": {
            "filter": [
                "icutokrep_no_camel_split",
                "lowercase",
                "suggest_shingle"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        },
        "suggest_reverse": {
            "filter": [
                "icutokrep_no_camel_split",
                "lowercase",
                "suggest_shingle",
                "reverse"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        },
        "text": {
            "char_filter": [
                "globo_norm",
                "acronym_fixer",
                "camelCase_splitter",
                "word_break_helper",
                "arabic_extended_norm"
            ],
            "filter": [
                "icu_token_repair",
                "homoglyph_norm",
                "lowercase",
                "decimal_digit",
                "arabic_extra_stop",
                "arabic_stop",
                "arabic_normalization",
                "arabic_stemmer",
                "asciifolding"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        },
        "text_search": {
            "char_filter": [
                "globo_norm",
                "acronym_fixer",
                "camelCase_splitter",
                "word_break_helper",
                "arabic_extended_norm"
            ],
            "filter": [
                "icu_token_repair",
                "homoglyph_norm",
                "lowercase",
                "decimal_digit",
                "arabic_extra_stop",
                "arabic_stop",
                "arabic_normalization",
                "arabic_stemmer",
                "asciifolding"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        },
        "token_reverse": {
            "filter": [
                "reverse"
            ],
            "tokenizer": "no_splitting",
            "type": "custom"
        },
        "trigram": {
            "filter": [
                "lowercase"
            ],
            "tokenizer": "trigram",
            "type": "custom"
        },
        "word_prefix": {
            "filter": [
                "icutokrep_no_camel_split",
                "lowercase",
                "prefix_ngram_filter"
            ],
            "tokenizer": "textify_icu_tokenizer",
            "type": "custom"
        }
    },
    "char_filter": {
        "arabic_extended_norm": {
            "type": "limited_mapping",
            "mappings": [
                "\\uFB8E=>\\u0643",
                "\\uFB8F=>\\u0643",
                "\\uFB90=>\\u0643",
                "\\uFB91=>\\u0643",
                "\\u06A9=>\\u0643",
                "\\u06AA=>\\u0643",
                "\\uFEDB=>\\u0643",
                "\\uFEDC=>\\u0643",
                "\\uFED9=>\\u0643",
                "\\uFEDA=>\\u0643",
                "\\uFBFC=>\\u064A",
                "\\uFBFD=>\\u064A",
                "\\uFBFE=>\\u064A",
                "\\uFBFF=>\\u064A",
                "\\u06CC=>\\u064A",
                "\\uFBE8=>\\u064A",
                "\\uFBE9=>\\u064A",
                "\\uFEEF=>\\u064A",
                "\\uFEF0=>\\u064A",
                "\\u0649=>\\u064A",
                "\\u06CD=>\\u064A",
                "\\uFBE4=>\\u064A",
                "\\uFBE5=>\\u064A",
                "\\uFBE6=>\\u064A",
                "\\uFBE7=>\\u064A",
                "\\u06D0=>\\u064A",
                "\\uFBA6=>\\u0647",
                "\\uFBA7=>\\u0647",
                "\\uFBA8=>\\u0647",
                "\\uFBA9=>\\u0647",
                "\\u06C1=>\\u0647",
                "\\u06C0=>\\u0647",
                "\\uFBA4=>\\u0647",
                "\\uFBA5=>\\u0647",
                "\\u06D5=>\\u0647"
            ]
        },
        "dotted_I_fix": {
            "type": "limited_mapping",
            "mappings": [
                "\u0130=>I"
            ]
        },
        "globo_norm": {
            "type": "mapping",
            "mappings": [
                "`=>'",
                "\u00b4=>'",
                "\u02b9=>'",
                "\u02bb=>'",
                "\u02bc=>'",
                "\u02bd=>'",
                "\u02be=>'",
                "\u02bf=>'",
                "\u02cb=>'",
                "\u055a=>'",
                "\\u05F3=>'",
                "\u2018=>'",
                "\u2019=>'",
                "\u201b=>'",
                "\u2032=>'",
                "\u2035=>'",
                "\ua78c=>'",
                "\uff07=>'",
                "\uff40=>'",
                "\\u202F=>\\u0020",
                "\u02c8=>",
                "\u02cc=>",
                "\\u0640=>",
                "\u066c=>,",
                "\u060c=>,",
                "\u055b=>",
                "\u055c=>",
                "\u055e=>",
                "\u00b5=>\u03bc",
                "\\u05F0=>\\u05D5\\u05D5",
                "\\u05F1=>\\u05D5\\u05D9",
                "\\u05F2=>\\u05D9\\u05D9",
                "\\uFB1F=>\\u05D9\\u05D9\\u05B7",
                "\\u05D9\\u05B7\\u05D9=>\\u05D9\\u05D9\\u05B7"
            ]
        },
        "near_space_flattener": {
            "type": "limited_mapping",
            "mappings": [
                "'=>\\u0020",
                "\\u2019=>\\u0020",
                "\\u02BC=>\\u0020",
                "_=>\\u0020",
                "-=>\\u0020"
            ]
        },
        "nnbsp_norm": {
            "type": "limited_mapping",
            "mappings": [
                "\\u202F=>\\u0020"
            ]
        },
        "regex_acronym_fixer": {
            "type": "pattern_replace",
            "pattern": "(?<=(?:^|\\P{L})\\p{L})[.\uff0e](\\p{L})(?=\\P{L}|$)",
            "replacement": "$1"
        },
        "regex_camelCase": {
            "type": "pattern_replace",
            "pattern": "(\\p{Ll}[\\p{M}\\p{Cf}]*)([\\p{Lu}\\p{Lt}])",
            "replacement": "$1 $2"
        },
        "word_break_helper": {
            "type": "limited_mapping",
            "mappings": [
                "_=>\\u0020",
                ":=>\\u0020",
                ".=>\\u0020",
                "(=>\\u0020",
                ")=>\\u0020",
                "\uff0e=>\\u0020",
                "\uff3f=>\\u0020",
                "\uff1a=>\\u0020",
                "\u00b7=>\\u0020"
            ]
        },
        "word_break_helper_source_text": {
            "type": "limited_mapping",
            "mappings": [
                "_=>\\u0020",
                ".=>\\u0020",
                "(=>\\u0020",
                ")=>\\u0020",
                ":=>\\u0020"
            ]
        }
    },
    "filter": {
        "aggressive_splitting": {
            "type": "word_delimiter_graph",
            "stem_english_possessive": false,
            "preserve_original": false
        },
        "arabic_extra_stop": {
            "type": "stop",
            "stopwords": [
                "\u0623\u0644\u064a",
                "\u0625\u0650\u0644\u064e\u0649",
                "\u0625\u0644\u0651\u0627",
                "\u0625\u0650\u0644\u064e\u0651\u0627",
                "\u0623\u0645\u0651\u0627",
                "\u0625\u0645\u0651\u0627",
                "\u0622\u0644\u0627\u0646",
                "\u0622\u0646",
                "\u0622\u0646\u0647",
                "\u0623\u0646\u0651",
                "\u0623\u064e\u0646\u064e\u0651",
                "\u0623\u064e\u0646\u0652",
                "\u0625\u0650\u0646\u064e\u0651",
                "\u0625\u0650\u0646\u064e\u0651\u0647\u064f",
                "\u0627\u0644\u0623\u0646",
                "\u0623\u0646\u062a\u0650",
                "\u0623\u064e\u0648\u0652",
                "\u0622\u0649",
                "\u0622\u064a",
                "\u0623\u064a\u0652",
                "\u0623\u064e\u064a\u0652",
                "\u0625\u0649",
                "\u0625\u064a",
                "\u0625\u064a\u0652",
                "\u0623\u064a\u0636\u0627\u064b",
                "\u0623\u064a\u0636\u064b\u0627",
                "\u0628\u0640",
                "\u0628\u0625\u0646",
                "\u0628\u062f\u0627\u0647",
                "\u0628\u062f\u0627\u0647\u0627",
                "\u0648\u0628\u062f\u0627",
                "\u0628\u0632\u0627\u0641",
                "\u0628\u0639\u0651\u062f",
                "\u0628\u0650\u0647\u0627",
                "\u0628\u0649",
                "\u0628\u064a\u0647",
                "\u0628\u064a\u0647\u0627",
                "\u0628\u064a\u0646\u0645\u0651\u0627",
                "\u062a\u0628\u0627\u0639",
                "\u0648\u062a\u0628\u0627\u0639",
                "\u0622\u0644\u062a\u0649",
                "\u062b\u064f\u0645\u064e\u0651",
                "\u062c\u0652\u0645\u064a\u0639",
                "\u062d\u062a\u0651\u0649",
                "\u062d\u064e\u064a\u0652\u062b\u064f",
                "\u062f",
                "\u0628\u0627\u0644\u062f\u0649",
                "\u062f\u0649",
                "\u062f\u064a",
                "\u062f\u064a\u0647",
                "\u062f\u064a\u0627\u0644",
                "\u062f\u064a\u0627\u0644\u0648",
                "\u0627\u0644\u064e\u0651\u0630\u0650\u064a",
                "\u0634\u0646\u0648",
                "\u0648\u0634\u0646\u0648",
                "\u0639\u064e\u0644\u064e\u0649",
                "\u0639\u0644\u064a",
                "\u0639\u0644\u064a\u0629",
                "\u0639\u064e\u0646\u0652",
                "\u0639\u0650\u0646\u0652\u062f\u064e",
                "\u063a\u064e\u064a\u0631",
                "\u063a\u064e\u064a\u0652\u0631\u064e",
                "\u0641\u0640",
                "\u0641\u0652\u0640",
                "\u0641\u0625\u0646",
                "\u0641\u064e\u0627\u0646",
                "\u0641\u0649",
                "\u0641\u0650\u064a\u0647\u0650",
                "\u0641\u0650\u064a\u0647\u064e\u0627",
                "\u0641\u064a\u064a",
                "\u0648\u0641\u064a\u0647",
                "\u0648\u0641\u064a\u0647\u0627",
                "\u0642\u062f\u0651",
                "\u0643\u064e\u0627\u0646\u064e",
                "\u0648\u064e\u0643\u064e\u0627\u0646\u064e",
                "\u0643\u0644\u0651",
                "\u0643\u064e\u0645\u064e\u0627",
                "\u0644",
                "\u0644\u0627\u064b",
                "\u0644\u0627\u0651",
                "\u0644\u064e\u0627",
                "\u0644\u0651\u0627",
                "\u0627\u0644\u0644\u0649",
                "\u0627\u0644\u0644\u064a",
                "\u0627\u0644\u0644\u064b\u064a",
                "\u0627\u0644\u0644\u0651\u064a",
                "\u0628\u0627\u0644\u0644\u0649",
                "\u0628\u0627\u0644\u0644\u064a",
                "\u0644\u0649",
                "\u0644\u064a",
                "\u0644\u0651\u064a",
                "\u0644\u064a\u0627\u0646",
                "\u0644\u064a\u0647",
                "\u0644\u064a\u0647\u0627",
                "\u0648\u0627\u0644\u0644\u0649",
                "\u0648\u0627\u0644\u0644\u064a",
                "\u0648\u0644\u064a\u0629",
                "\u0645\u064e\u0627",
                "\u0645\u064e\u0646\u0652",
                "\u0645\u0650\u0646",
                "\u0645\u0650\u0646\u064e",
                "\u0645\u0650\u0646\u0652",
                "\u0645\u0646\u0629",
                "\u0645\u0650\u0646\u0652\u0647\u064f",
                "\u0645\u0646\u064b\u0647\u0627",
                "\u0645\u0646\u0651\u0647\u0627",
                "\u0647\u0627\u062f",
                "\u0647\u0648\u0651",
                "\u0647\u064f\u0648\u064e",
                "\u0647\u0648\u0651\u0627",
                "\u0647\u064a\u0651",
                "\u0647\u0650\u064a",
                "\u0624",
                "\u0648\u0641\u0649",
                "\u0648\u064f\u0643\u0650\u0651\u0644\u064e",
                "\u0644\u0644\u0648\u0644\u0627",
                "\u0648\u0644\u0623\u0651",
                "\u0648\u0644\u0651\u0623",
                "\u0648\u0644\u0627\u064b",
                "\u0648\u0644\u0627\u064e",
                "\u0648\u0644\u0627\u064f",
                "\u0648\u0644\u0627\u0651",
                "\u0648\u0644\u0651\u0627",
                "\u0648\u0652\u0644\u0627",
                "\u0648\u0648\u0644\u0627",
                "\u0648\u0648\u0644\u0627\u064b",
                "\u0648\u0652\u0645\u0646",
                "\u0648\u0647\u0650\u064a",
                "\u0648\u064e\u0647\u064f\u0648\u064e",
                "\u0641\u0649",
                "\u0641\u06cc\u0647\u0627",
                "\u0648\u0641\u0649",
                "\u062d\u062a\u0649",
                "\u0639\u0644\u0649",
                "\u0644\u062f\u0649"
            ]
        },
        "arabic_stemmer": {
            "type": "stemmer",
            "language": "arabic"
        },
        "arabic_stop": {
            "type": "stop",
            "stopwords": "_arabic_"
        },
        "asciifolding": {
            "type": "asciifolding",
            "preserve_original": false
        },
        "asciifolding_preserve": {
            "type": "asciifolding",
            "preserve_original": true
        },
        "dedup_asciifolding": {
            "type": "unique",
            "only_on_same_position": true
        },
        "icutokrep_no_camel_split": {
            "type": "icu_token_repair",
            "keep_camel_split": false
        },
        "lowercase": {
            "type": "lowercase"
        },
        "prefix_ngram_filter": {
            "type": "edgeNGram",
            "max_gram": 255
        },
        "remove_empty": {
            "type": "length",
            "min": 1
        },
        "suggest_shingle": {
            "type": "shingle",
            "min_shingle_size": 2,
            "max_shingle_size": 3,
            "output_unigrams": true
        },
        "truncate_keyword": {
            "type": "truncate",
            "length": 5000
        }
    },
    "tokenizer": {
        "no_splitting": {
            "type": "keyword"
        },
        "prefix": {
            "type": "edgeNGram",
            "max_gram": 255
        },
        "trigram": {
            "type": "nGram",
            "min_gram": 3,
            "max_gram": 3
        }
    }
}